src/voting-tools-rs/src/testing/test_api/output_assertions.rs

```diff
@@ -0,0 +1,70 @@
+#![allow(dead_code)]
+
+use crate::model::Output;
+use bigdecimal::ToPrimitive;
+use cardano_serialization_lib::address::RewardAddress;
+use cardano_serialization_lib::crypto::PublicKey;
+
+/// Allows [Output] struct to be assertable
+pub trait VerifiableSnapshotOutput {
+    /// returns assertion struct
+    fn assert(&self) -> SnapshotOutputAssert;
+}
+
+impl VerifiableSnapshotOutput for Output {
+    fn assert(&self) -> SnapshotOutputAssert {
+        SnapshotOutputAssert::new(self)
+    }
+}
+
+/// Fluent api for [Output] assertions
+pub struct SnapshotOutputAssert<'a> {
+    output: &'a Output,
+}
```

Comment:
This seems like quite a lot of infrastructure for asserting stuff. Does this do something that a simple `assert_eq!()` in the test code wouldn't do?

Reply:
Yes, again this is investment for the future as even now i have plan to add more test cases so this will avoid duplicating those assertions in other part of the repo

Reply:
Similar to above, I usually introduce traits/abstractions when they're needed by the code, but happy to go with this

Reply:
ok, thanks!

Reply:
Hm, i think that reviewer should resolve each conversation so that he accept my explanation or fix. But yes i can do it too ;)